lib/api/initiateMultipartUpload.js

```diff
@@ -136,30 +139,47 @@ function initiateMultipartUpload(authInfo, request, log, callback) {
                 masterKeyId: serverSideEncryption.masterKeyId,
             };
         }
-        services.getMPUBucket(destinationBucket, bucketName, log,
-            (err, MPUbucket) => {
-                // BACKWARD: Remove to remove the old splitter
-                if (MPUbucket.getMdBucketModelVersion() < 2) {
-                    metadataStoreParams.splitter = constants.oldSplitter;
-                }
-                services.metadataStoreMPObject(MPUbucket.getName(),
-                    cipherBundle, metadataStoreParams,
-                    log, err => {
-                        if (err) {
-                            log.trace('error storing multipart object', {
-                                error: err,
+        return locationConstraintCheck(request, null, destinationBucket, log,
+        (err, backendInfo) => {
+            if (err) {
+                return process.nextTick(() => {
+                    callback(err.customizeDescription('Location Constraint ' +
+                    'Info is invalid.'));
+                });
+            }
+            if (config.regions) {
+                metadataStoreParams.controllingLocationConstraint =
+                   backendInfo;
```

Comment:
can you just return the locationConstraint so you can treat these two assignments the same?